.github/workflows/go.yml

```diff
@@ -9,20 +9,23 @@ jobs:
     if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
 
     steps:
-      - name: Set up Go 1.16
-        uses: actions/setup-go@v1
+      - name: Set up Go 1.18
+        uses: actions/setup-go@v3
         with:
-          go-version: 1.16
+          go-version: 1.18
         id: go
 
       - name: Check out code into the Go module directory
-        uses: actions/checkout@v2
-
-      - name: Install golanglint-ci
-        run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.37.1
+        uses: actions/checkout@v3
 
       - name: lint
-        run: PATH=$(go env GOPATH)/bin:$PATH make lint
+        uses: golangci/golangci-lint-action@v3
```

Comment:
Swapped this with the actual golang ci lint action so it can also pop lint error messages on PRs.

Reply:
Good idea!